model_compression_toolkit/core/keras/hessian/weights_trace_hessian_calculator_keras.py

```diff
@@ -50,5 +53,112 @@ def __init__(self,
                                                                  trace_hessian_request=trace_hessian_request,
                                                                  num_iterations_for_approximation=num_iterations_for_approximation)
 
-    def compute(self):
-        Logger.error(f"Hessian trace approx w.r.t weights is not supported for now")
\ No newline at end of file
+    def compute(self) -> np.ndarray:
+        """
+        Compute the Hessian-based scores for w.r.t target node's weights.
+        Currently, supported nodes are [Conv2D, Dense, Conv2DTranspose, DepthwiseConv2D].
+        The computed scores are returned in a numpy array. The shape of the result differs
+        according to the requested granularity. If for example the node is Conv2D with a kernel
+        shape of (3, 3, 3, 2) (namely, 3 input channels, 2 output channels and kernel size of 3x3)
+        and the required granularity is HessianInfoGranularity.PER_TENSOR the result shape will be (1,),
+        for HessianInfoGranularity.PER_OUTPUT_CHANNEL the shape will be (2,) and for
+        HessianInfoGranularity.PER_ELEMENT a shape of (3, 3, 3, 2).
+
+        Returns:  The computed scores as numpy ndarray.
+
+        """
+        # Check if the target node's layer type is supported
+        if self.hessian_request.target_node.layer_class not in [Conv2D, Dense, Conv2DTranspose, DepthwiseConv2D]:
+            Logger.error(
+                f"{self.hessian_request.target_node.type} is not supported for Hessian info w.r.t weights.")
```

Comment:
I wonder if we want to return error for every layer we don't support like you did or just for layer with weights we don't support. For now it means someone should walk over all layers and pick only the ones with weights that we support in order no to get an error (like you did in the tests). What if when there is no weight (like relu) return None?

Reply:
I think that this might lead to parts of the code frequently checking for None values, which could make the code hard to read and maintain. I think it's reasonable that unsupported requests will trigger an error, ensuring clarity of usage and providing direct feedback about the exact problem.